### ISSUE TITLE
HMR: retain state not working?

### ISSUE THREAD

**Issue by renaudcerrato** [rjaros/kvision#186]
Hi there!

Using 3.16.0, I'm playing with KVision, and I'm unable to get my state back during `fun start(state: Map<String,Any>)` despite having overriden `dispose()` : I'm always getting an empty map. I'm a Senior Android Developer, but a frontend newbie - am I missing something?

```kotlin
@Serializable
data class State(val clicks: Int = 0, val date: String = "")

val store = ObservableValue(State())
...
override fun dispose(): Map<String, Any> = 
  mapOf("state" to store.value)

```

I also tried `toObj()` without success:

```kotlin
override fun dispose(): Map<String, Any> = 
  mapOf("state" to store.value.toObj())

```

Of course hot reloading is working perfectly fine, apart from that. Am I missing something?

**Comment by rjaros** [comment]
Hello,
I confirm it doesn't seem to work as expected. Probably something has changed in the HMR architecture. I'll investigate and let you know.

**Comment by rjaros** [comment]
Obviously we have lost HMR some time ago during the migration from old `kotlin-frontend-plugin` to the current Kotlin/JS plugin. I haven't even realized HRM is not the same as Live Reload, and current configuration works with Live Reload by default.

But I've managed to make it work again. Unfortunately some small fixes are required in KVision code, so it will be supported with the next release. I'll make changes in the guide and fix all example applications - it will require passing additional parameter to the `startApplication` function, like this:
```
fun main() {
    startApplication(::App, module.hot)
}
```
And the application reload cycle with HMR is at least twice as fast compared to the default configuration!

**Comment by rjaros** [comment]
Fixed in 3.16.1.

Fully working example below:
```
package com.example

import pl.treksoft.kvision.Application
import pl.treksoft.kvision.html.button
import pl.treksoft.kvision.html.div
import pl.treksoft.kvision.module
import pl.treksoft.kvision.panel.root
import pl.treksoft.kvision.panel.vPanel
import pl.treksoft.kvision.require
import pl.treksoft.kvision.startApplication
import pl.treksoft.kvision.state.ObservableValue

data class State(val clicks: Int = 0, val date: String = "")

val store = ObservableValue(State())

class App : Application() {

    override fun start(state: Map<String, Any>) {
        (state["state"]?.unsafeCast<State>())?.let {
            store.value = it
        }
        root("kvapp") {
            vPanel(spacing = 5) {
                div(store) { state ->
                    div("Number of clicks: ${state.clicks}")
                }
                button("Click").onClick {
                    store.value = store.value.copy(clicks = store.value.clicks + 1)
                }
            }
        }
    }

    override fun dispose(): Map<String, Any> {
        return mapOf("state" to store.value)
    }
}

fun main() {
    startApplication(::App, module.hot)
}
```
